### PR TITLE
Added  CspInvocation and CspResponse objects

### PR DESCRIPTION
**Pull request by jeffsegal-ccpo**
The CSP Provisioning process will be executed roughly as follows:

1. User interacts with ATAT UI / Internal API to update ATAT resources (Portfolios, Applications, Environments & Task Orders)
2. When ready, User submits all pending resources for provisioning. Pending resources are defined as new rows in the database whose status is not `COMPLETE` or `IN_PROGRESS`
3. The POST /portfolios/:idsubmit Lambda walks the tree of the given Portfolio and creates one or more `CspInvocation` objects and passing each to their own instance of a Step Function workflow
4. The  CSP Invocation Lambda reads the given `CspInvocation` object and makes a matching invocation to the endpoint contained within it (initial implementation in Phase 1 will be totally mocked out, but include ability to test all paths)
    1.  If a 5xx code is received, retry up to 6 times
    2. If a 4xx code is received, invoke the Reject CSP Response Lambda. This will mark each resource under the parent with `provisioning_status` = `FAILED`.
    3. If a 2xx code is received, invoke the Persist CSP Response Lambda. This will mark each resource under the parent with `provisioning_status` = `COMPLETE`.

We'll need a proper formal schema In order to properly exchange information between these Lambdas. It should also facilitate parallel development of these Lambdas. If any other objects need to be created, they can be added here later. In theory we could add them to a different spec file but this seems fine for now. 

Ticket: AT-6707 (CSP Provisioning Epic)

### CODE REVIEW

File: atat_internal_api.yaml [hunk 1]
```diff
@@ -2038,6 +2038,37 @@ components:
...
+          format: uri
+        payload:
+          type: object
+    CspResponse:
+      properties:
+        code:
+          type: integer
+        content:
+          type: object
+            additionalProperties: true
```
Comment (zachclark-ccpo) [inline atat_internal_api.yaml:2069]: `additionalProperties: true` is true by default, could probably get away with removing it.

File: atat_internal_api.yaml [hunk 2]
```diff
@@ -2038,6 +2038,37 @@ components:
...
+        method:
+          type: string
+          enum:
+            - GET
+            - POST
+            - PUT
+            - DELETE
+        headers:
+          type: object
+          additionalProperties:
```
Comment (zachclark-ccpo) [inline atat_internal_api.yaml:2056]: I was a bit confused seeing additionalProperties used this, but it checks out.
Link to docs if anyone is interested: https://swagger.io/docs/specification/data-models/dictionaries/

File: atat_internal_api.yaml [hunk 3]
```diff
@@ -2038,6 +2038,37 @@ components:
   #          type: string
   #        lastName:
   #          type: string
+    CspInvocation:
+      required:
+        - method
+        - endpoint
+        - payload
```
Comment (zachclark-ccpo) [inline atat_internal_api.yaml:2045]: Are `headers` not required?

File: atat_internal_api.yaml (hunk 2 above)
Comment (jeffsegal-ccpo) [reply atat_internal_api.yaml:2056]: Yeah - I believe this is the best practice for defining an object as a `string` -> `string` map (i.e. what we would want for HTTP Headers).

File: atat_internal_api.yaml (hunk 3 above)
Comment (jeffsegal-ccpo) [reply atat_internal_api.yaml:2045]: Nope - I didn't want to impose anything on the process but in theory we might need to pass some headers in some day that might require some dynamic calculation or other reason for including in this object (vs just hard-coding it in the invocation Lambda).

**Review by zachclark-ccpo — APPROVED** [review]

File: atat_internal_api.yaml (hunk 2 above)
Comment (jasonburkert-ccpo) [reply atat_internal_api.yaml:2056]: Thanks for linking those docs

File: atat_internal_api.yaml (hunk 1 above)
Comment (zachclark-ccpo) [inline atat_internal_api.yaml:2069]:
```suggestion
        content:
          type: object
          additionalProperties: true
```

Comment (zachclark-ccpo) [reply atat_internal_api.yaml:2069]: Nested mapping fix. from @jasonburkert-ccpo  slack comment

**Review by jasonburkert-ccpo — APPROVED** [review]
Looks good after

**Pull request merged**